app/pods/publications/publication/documents/controller.js

```diff
@@ -57,4 +83,131 @@ export default class PublicationDocumentsController extends Controller {
   toggleFolderCollapse(piece) {
     piece.set('collapsed', !piece.collapsed);
   }
+
+  @action
+  uploadPiece(file) {
+    const now = moment().utc()
+      .toDate();
+    const documentContainer = this.store.createRecord('document-container', {
+      created: now,
+    });
+    const piece = this.store.createRecord('piece', {
+      created: now,
+      modified: now,
+      file: file,
+      accessLevel: this.defaultAccessLevel,
+      confidential: false,
+      name: file.filenameWithoutExtension,
+      documentContainer: documentContainer,
+    });
+    this.newPieces.pushObject(piece);
+  }
+
+  @task
+  *savePieces() {
+    const savePromises = this.newPieces.map(async(piece) => {
+      try {
+        this.savePiece.perform(piece);
+      } catch (error) {
+        this.deletePiece.perform(piece);
+        throw error;
+      }
+    });
+    yield all(savePromises);
+    this.isOpenPieceUploadModal = false;
+    this.newPieces = A();
+  }
+
+  /**
+   * Save a new document container and the piece it wraps
+   */
+  @task
+  *savePiece(piece) {
+    const documentContainer = yield piece.documentContainer;
+    yield documentContainer.save();
+    yield piece.save();
+    const pieces = yield this.model.case.hasMany('pieces').reload();
+    pieces.pushObject(piece);
+    yield this.model.case.save();
+  }
+
+  @task
+  *cancelUploadPieces() {
+    const deletePromises = this.newPieces.map((piece) => this.deletePiece.perform(piece));
+    yield all(deletePromises);
+    this.newPieces = A();
+    this.isOpenPieceUploadModal = false;
+  }
+
+  @task
+  *deletePiece(piece) {
+    const file = yield piece.file;
+    yield file.destroyRecord();
+    this.newPieces.removeObject(piece);
+    const documentContainer = yield piece.documentContainer;
+    yield documentContainer.destroyRecord();
+    yield piece.destroyRecord();
+  }
+
+
+  /** TRANSLATION ACTIVITIES **/
+
+  @action
+  openTranslationRequestModal() {
+    this.translateActivity.finalTranslationDate = this.model.publicationFlow.translateBefore;
+    this.translateActivity.pieces = this.selectedPieces;
+    this.showTranslationModal = true;
+  }
+
+  get getTranslateActivityBeforeDate() {
+    if (this.model.publicationFlow.translateBefore) {
+      return this.model.publicationFlow.translateBefore;
+    }
+    return new Date();
```

Comment:
We zijn zeker dat dit ook de `utc` tijd neemt by default?

Reply:
1. Dit gaat zelden voorkomen aangezien een publication meestal een final translation date zal hebben, en dit dus niet gebruikt zal worden
2. dit gaat zowiezo aangepast worden door de gebruiker in het geval bovenstaande niet waar is
m.a.w. de default waarde is eigenlijk onbelangrijk.
Maar of fit utc gebruikt? not sure.
@brechtPhilips ?

Reply:
Hmm maar dan vraag ik me af.. heeft het dan zin dat we een `new Date` doen.. want in essentie tonen we dan de datum van vandaag.. is dat de bedoeling? Misschien eerder leeg laten dan en een placeholder voorzien? Als het model bv per ongeluk niet goed geset is.. dan krijgen we de datum van vandaag te zien waardoor we misschien denken dat de datum ook effectief op vandaag stond.. lijkt me een nasty bug om te vinden..

Reply:
Ik zou dit meepakken en eventueel in een demo de vraag stellen.
Er zijn verschillende assumpties die gemaakt worden. 
Zowiezo zal er links en rechts nog moeten gesleuteld worden.
In je opmerking spreek je over het model dat niet gesaved wordt? 
Het gaat over een item dat nog niet bestaat. He t moet nog aangemaakt worden.

Reply:
Ok